misk/src/test/resources/test_data_source_app-common.yaml

```diff
@@ -0,0 +1,7 @@
+data_sources:
+  databases:
```

Comment:
it isn't great to have this additional level of nesting. I think we'd need to update the `DataSourcesConfig` class to implement the `Map` interface so Jackson populates it properly (?).

Reply:
yeah totally agreed; can it just implement Map, or do we need a custom Yaml unmarshaller for that type?

Reply:
Not 100% sure, I'm going to investigate a bit and will follow up with another PR